aspen/http/resource.py

```diff
@@ -7,25 +8,18 @@ class Static(object):
     """
 
     def __init__(self, request_processor, fspath, raw, media_type):
-        self.request_processor = request_processor
+        assert type(raw) is bytes  # sanity check
         self.raw = raw
         self.media_type = media_type
         if media_type == 'application/json':
-            self.media_type = self.request_processor.media_type_json
+            self.media_type = request_processor.media_type_json
+        if self.media_type.startswith('text/'):  # XXX this is bad
```

Comment:
> In that case maybe we should just drop `charset_static` entirely, and let users implement their own algorithm function for varying the charset of static files if they want.

An algorithm function is executed for each request, it can't do something that should only be done once for each `Static` resource without adding a small cost.

> Or, we could make `charset` a property of the `resource.{Static,Dynamic}` classes instead of a global configuration value.

I think that would be a step backwards as I'm trying to get rid of `charset` attributes that mean "this **should be** encoded with this codec" instead of "this **is** encoded with this charset". (Pando's `Response.charset` means "should encode with", I'm planning to remove it eventually.)